### PR TITLE
Update release notes for 0.3

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Updates release notes from the PRs tagged as `kind/feature` or `kind/bug` that are user-facing.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Let me know if I missed a PR.

In the future we should be more diligent adding release notes (I'm guilty too). And we should probably add milestones too.

### CODE REVIEW

**Comment by kannon92** [comment]
> cc @kannon92 @kerthcet @tenzen-y @mimowo as people who worked on lots of PRs during this release

Congratulations! Sounds like a lot of good things in this release!

**Comment by tenzen-y** [comment]
Congrats!

File: CHANGELOG/CHANGELOG-0.3.md [hunk 1]
```diff
@@ -4,12 +4,55 @@ Changes since `v0.2.1`:
...
+  `v1beta1` includes the following changes:
+  - Add `namespace` to propagate the namespace where kueue is deployed to the webhook certificate.
+  - Add `internalCertManagement` with fields `enable`, `webhookServiceName` and `webhookSecretName`.
+  - Remove `enableInternalCertManagement`. Use `internalCertManagement.enable` instead.
+- Upgrade the `kueue.x-k8s.io` API version from `v1alpha2` to `v1beta1`.
+  `v1alpha2` is no longer supported.
+  `v1beta1` includes the following changes:
+  - `ClusterQueue`:
+    - Immutability of `spec.queueingStrategy`.
+    - Refactor `quota.min` and `quota.max` into `nominalQuota and `borrowingLimit`.
```
Comment (tenzen-y) [inline CHANGELOG/CHANGELOG-0.3.md:18]:
```suggestion
    - Refactor `quota.min` and `quota.max` into `nominalQuota` and `borrowingLimit`.
```

**Review by tenzen-y — COMMENTED** [review]
Thank you for updating the CHANGELOG.
I left a comment for a nit.

File: CHANGELOG/CHANGELOG-0.3.md [hunk 2]
```diff
@@ -4,12 +4,55 @@ Changes since `v0.2.1`:
...
+    - Rename `taints` to `nodeTaints`.
+- Obtain queue names from label `kueue.x-k8s.io/queue-name`. The annotation with
+  the same name is still supported, but it's now deprecated.
+- Multiplatform support for `linux/amd64` and `linux/arm64`.
+- Validating webhook for `batch/v1.Job` validates kueue-specific labels and
+  annotations.
+- Use of Server-Side-Apply for updating conditions.
+- Sequential admission of jobs https://kueue.sigs.k8s.io/docs/tasks/setup_sequential_admission/
+- Preemption within ClusterQueue and cohort https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/#preemption
+- Library for integrating job-like CRDs (controller and webhooks) https://sigs.k8s.io/kueue/pkg/controller/jobframework
```
Comment (kerthcet) [inline CHANGELOG/CHANGELOG-0.3.md:41]: limitRange https://github.com/kubernetes-sigs/kueue/issues/541 should be included.

File: CHANGELOG/CHANGELOG-0.3.md [hunk 3]
```diff
@@ -4,12 +4,55 @@ Changes since `v0.2.1`:
...
+  - `ResourceFlavor`:
+    - Introduce `spec` to hold all fields.
+    - Rename `labels` to `nodeLabels`.
+    - Rename `taints` to `nodeTaints`.
+- Obtain queue names from label `kueue.x-k8s.io/queue-name`. The annotation with
+  the same name is still supported, but it's now deprecated.
+- Multiplatform support for `linux/amd64` and `linux/arm64`.
+- Validating webhook for `batch/v1.Job` validates kueue-specific labels and
+  annotations.
+- Use of Server-Side-Apply for updating conditions.
```
Comment (kerthcet) [inline CHANGELOG/CHANGELOG-0.3.md:38]: I think we can remove it as it's not user-facing.

File: CHANGELOG/CHANGELOG-0.3.md (hunk 2 above)
Comment (kerthcet) [reply CHANGELOG/CHANGELOG-0.3.md:41]: Also support scheduler leader election for HA.

File: CHANGELOG/CHANGELOG-0.3.md [hunk 4]
```diff
@@ -4,12 +4,55 @@ Changes since `v0.2.1`:
...
+- Reproducible load tests https://sigs.k8s.io/kueue/test/performance
+- Documentation website https://kueue.sigs.k8s.io/docs/
 
 ### Bug fixes
 
-- Fix job controller ClusterRole for clusters that enable OwnerReferencesPermissionEnforcement admission control validation
+- Fix job controller ClusterRole for clusters that enable OwnerReferencesPermissionEnforcement admission control validation #392
+- Fix race condition when admission attempt and requeuing happen at the same time #427
+- Atomically release quota and requeue previously inadmissible workloads #512
+- Fix support for leader election #580
```
Comment (kerthcet) [inline CHANGELOG/CHANGELOG-0.3.md:58]: Oh, you added here, maybe more like a feature?

File: CHANGELOG/CHANGELOG-0.3.md (hunk 2 above)
Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-0.3.md:41]: Added LimitRange

**Comment by tenzen-y** [comment]
Thanks!
/lgtm
/hold for @kerthcet

**Comment by kerthcet** [comment]
/lgtm
/hold cancel

Thanks @alculquicondor

**Pull request merged**